zephyr/CMakeLists.txt

```diff
@@ -37,6 +37,7 @@ file(GLOB_RECURSE Sources
   "../src/session/*.c"
   "../src/transport/*.c"
   "../src/utils/*.c"
+  "../src/system/*.c"
```

Comment:
As this is flagged as GLOB_RECURSE it includes all the subdirectories....lots of odd errors from other platforms.

Try adding to the next list like:
```
file (GLOB Sources_Zephyr 
"../src/system/zephyr/*.c"
"../src/system/*.c"
)
```